lab3/oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -39,9 +44,11 @@ public void move(MoveDirection direction){
             case RIGHT:
                 orientation = orientation.next();
                 break;
+
             case LEFT:
                 orientation = orientation.previous();
                 break;
+
             case FORWARD:
                 Vector2d newPosition = position.add(orientation.toUnitVector());
 
```

Comment:
zabezpieczenie przed wyjściem z mapy powtarza się w dwóch miejscach - można to wyciągnąć poza switcha, jeśli zadeklarujemy `newPosition` wcześniej